### PR TITLE
Add missing pull_request id env variable

### LINKED ISSUE #52: Incorrect handling of PR builds from GitHub Actions.
Summary from the customer:

>  I did a bit of looking today. It appears that `pull_request` events use an automatic merge commit for the pull request, and the GITHUB_SHA is showing that commit hash instead of the one for the commit on the branch that was actually pushed. It makes some sense, but it's a big odd to get your head around. I believe that Travis CI does the same thing, except that they make it easy to know what the underlying commit you're trying to test about is, but GitHub actions does not expose that in an environment variable.

>In order to trace it back to the commit we're really trying to identify, we need to look at the parents of the commit we're on. I'm not sure the optimal way to do that, but the commit message of the current commit has it, and of course git itself has record of what the parent commits are as well, though I'm not totally sure the best way to figure out which of the parents is really the one we're looking for.

It seems that GitHub Actions may make a merge commit build for PRs, similar to how Travis CI does. I'm not sure if they commit this SHA, but the Codecov action does not use the correct SHA in this case.

### PR DESCRIPTION
**Pull request by armano2**
I did small refactoring of code as everything was typed there 4 times

cc @drazisil @ibrahim0814 
fixes #52

### CODE REVIEW

**Comment by ibrahim0814** [comment]
Hi @armano2! Thank you for taking the time to make this pull request! Unfortunately, the changes you made need to be reflected on our [bash uploader](github.com/codecov/codecov-bash) instead of in this repository. I've made a similar PR on our bash uploader that fixes incorrect PR handling for GitHub Actions here: https://github.com/codecov/codecov-bash/pull/263

For this repository, the only change that needs to be made is to add the GITHUB_HEAD_REF environment variable if it exists.

**Comment by armano2** [comment]
ok i will update PR, thank you for feedback

**Comment by armano2** [comment]
@ibrahim0814 updated, i'm unsure why ncc outputs packages in different order in dist

File: index.js [hunk 1]
```diff
@@ -55,101 +55,43 @@ try {
         GITHUB_ACTION: process.env.GITHUB_ACTION,
         GITHUB_REF: process.env.GITHUB_REF,
         GITHUB_REPOSITORY: process.env.GITHUB_REPOSITORY,
-        GITHUB_SHA: process.env.GITHUB_SHA
+        GITHUB_SHA: process.env.GITHUB_SHA,
+        GITHUB_HEAD_REF: process.env.GITHUB_HEAD_REF
```
Comment (ibrahim0814) [inline index.js]: Any way we can check to see if it exists first? Wouldn't this throw an error according to these docs: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/using-environment-variables#default-environment-variables

Comment (armano2) [reply index.js]: error? there will not be a error its just going to result undefined (same for other env variables)

Comment (armano2) [reply index.js]: i changed it to
```js
GITHUB_HEAD_REF: process.env.GITHUB_HEAD_REF || ''
```

**Comment by ibrahim0814** [comment]
Going to be adding a few commits to your changes here @armano2! Currently removing the `yml` parameter since we're having a ton of issues with yaml validation on our end. Nothing to do with your changes but thought I'd roll it in together since I want to release a new version of this action soon

**Comment by ibrahim0814** [comment]
Will also be adding in documentation and code to accommodate tokenless functionality for Actions in this PR

**Pull request merged**